frontend/src/theme/theme.ts

```diff
@@ -0,0 +1,241 @@
+import {
+  createTheme,
+  PaletteOptions,
+  SimplePaletteColorOptions,
+  TypographyOptions,
+} from "@mui/material/styles";
+import { TypographyStyleOptions } from "@mui/material/styles/createTypography";
+import { error, learner, administrator, facilitator } from "./palette";
+import "@fontsource/roboto";
+
+// adding custom attributes to palette
+declare module "@mui/material/styles" {
+  // allow configuration using `createTheme`
+  interface PaletteOptions {
+    // Figma: primary colour palette
+    learner: SimplePaletteColorOptions;
+
+    // Figma: secondary colour palette
+    administrator: SimplePaletteColorOptions;
+
+    // Figma: tertiary colour palette
+    facilitator: SimplePaletteColorOptions;
+  }
+
+  interface TypographyOptions {
+    // Figma: typography section
+    displayLarge?: TypographyStyleOptions;
+    displayMedium?: TypographyStyleOptions;
+    displaySmall?: TypographyStyleOptions;
+    headlineLarge?: TypographyStyleOptions;
+    headlineMedium?: TypographyStyleOptions;
+    headlineSmall?: TypographyStyleOptions;
+    titleLarge?: TypographyStyleOptions;
+    titleMedium?: TypographyStyleOptions;
+    titleSmall?: TypographyStyleOptions;
+    labelLarge?: TypographyStyleOptions;
+    labelMedium?: TypographyStyleOptions;
+    labelSmall?: TypographyStyleOptions;
+    bodyLarge?: TypographyStyleOptions;
+    bodyMedium?: TypographyStyleOptions;
+    bodySmall?: TypographyStyleOptions;
+  }
+
+  interface Theme {
+    status: {
+      danger: string;
```

Comment:
i assume that status is something like "warning" | "success" | "error", so maybe we can replace string with a custom status type instead